### PR TITLE
added highpass and unsharp_mask

### PR DESCRIPTION
**Pull request by alexlib**
also verified that unsharp_mask is equivalent to fast_box_blur at least for n=1,2
includes tests

### CODE REVIEW

**Comment by yosefm** [comment]
fast_ox_blur is just unsharp_mask cleaned up. I explained that in PR#53.

**Comment by alexlib** [comment]
should we still add highpass that uses fast_box_blur? it's 2 min work.

**Comment by yosefm** [comment]
In my repo I replaced copy_images with a simple memcpy. Perhaps it's good to have a copy_images function so we can transparently extend to 16 bit without user knowing, but it better use memcpy() internally.

Also, what is the use of histogram and histeq?

**Comment by alexlib** [comment]
in the past we had something that was called "enhance" or "histogram equalization" to improve the contrast in the image. It was used before highpass as far as I remember. i do not see harm in having it back, with test, in liboptv

File: liboptv/src/image_processing.c [hunk 1]
```diff
@@ -276,3 +276,304 @@ void split(unsigned char *img, int half_selector, control_par *cpar) {
...
+* subtract_mask compares img with img_mask and creates a masked image img_new
+* pixels that are equal to zero in the img_mask
+* are overwritten with a default value (=0) in img_new
+*  Arguments:
+*      img  is the unsigned char array pointers to the original image
+%      img  is the unsigned char array pointers to the mask (0 == mask out)
+*      img_new is the pointer to the unsigned char array for the resulting image
+*      control_par *cpar - contains image size parameters.
+*/
+void subtract_mask (unsigned char *img, unsigned char *img_mask, unsigned char *img_new, control_par *cpar)
```
Comment (yosefm) [inline liboptv/src/image_processing.c]: Please fix indentations and make lines <80 chars. Thanks.

File: liboptv/src/image_processing.c [hunk 2]
```diff
@@ -276,3 +276,304 @@ void split(unsigned char *img, int half_selector, control_par *cpar) {
...
+	int image_size = cpar->imx * cpar->imy;
+	
+	for (end=img1+image_size, ptr1=img1, ptr2=img2; ptr1<end; ptr1++, ptr2++)
+	*ptr2 = *ptr1;
+}
+
+/* histogram computes a histogram from the image and returns it to hist
+*  Arguments:
+*      img  is the unsigned char array pointer to the image array, 8 bit 
+*      control_par *cpar - contains image size parameters.
```
Comment (yosefm) [inline liboptv/src/image_processing.c]: Need to specify that the return value hist must have place for 256 ints.

**Comment by yosefm** [comment]
The problem with adding unused functions is that you later have to maintain them and change each one with every change to the interface. However, if you really want it and there's a test, I will not strongly object.

fast_box_blur() is the only version that should go in. unsharp_mask() should be deleted. highpass() should be made to use fast_box_blur(). I was working on that but you beat me to it :)

**Comment by alexlib** [comment]
Ok, i hope it's all fixed now:
highpass is using fast_box_blur
indentations, lines, comments
histeq and histogram might stay - will be removed if not used (but i think it is used now in Tcl/Tk and they might decide to interface liboptv at some point of time. for whatever reason IfU at ETH Zurich is still using this software)

**Comment by alexlib** [comment]
I think that having separate and clear copy_images() makes sense

File: liboptv/tests/check_image_proc.c [hunk 3]
```diff
@@ -141,9 +141,7 @@ END_TEST
 
 START_TEST(test_split)
 {
-    /*  A 3x3 box-blur is equivalent to lowpass_3, so that's the comparison
```
Comment (yosefm) [inline liboptv/tests/check_image_proc.c]: Why is this removed?

Comment (alexlib) [inline liboptv/tests/check_image_proc.c]: my mistake.

Comment (alexlib) [inline liboptv/tests/check_image_proc.c]: isn't it a copy/paste of the comment from test_blur and doesn't relate to test_split ?

Comment (yosefm) [inline liboptv/tests/check_image_proc.c]: You're right. Well, it's also present in PR55 so all is well :)
